app/src/main/java/co/nimblehq/compose/crypto/extension/BigDecimalExt.kt

```diff
@@ -0,0 +1,20 @@
+package co.nimblehq.compose.crypto.extension
+
+import java.math.BigDecimal
+import java.text.DecimalFormat
+
+fun BigDecimal.toFormattedString(
+    format: String = "#,###",
+    minDigit: Int = 0,
+    maxDigit: Int = 2
+): String {
+    return try {
+        DecimalFormat(format).apply {
+            minimumIntegerDigits = 1
+            minimumFractionDigits = minDigit
+            maximumFractionDigits = maxDigit
+        }.format(this)
+    } catch (e: IllegalArgumentException) {
+        ""
```

Comment:
Should we return "0" 🤔

Reply:
@Wadeewee this is the format func, I don't think 0 make sense here in case of error.

Reply:
Hmm 🤔 But we will call in the `UI` screen. Is that mean when we have an error, we will show just `$` or `%`?

Reply:
@Wadeewee Thanks for raising, that's an edge case, we should take care of that. Let's do it later, we don't have much time to do everything in just 1 PR 🙏